indexing-service/src/main/java/org/apache/druid/indexing/common/actions/RetrieveUpgradedToSegmentsIdsAction.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.common.actions;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.common.collect.ImmutableList;
+import org.apache.druid.indexing.common.task.Task;
+import org.apache.druid.metadata.SegmentUpgradeInfo;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Task action to retrieve the segment upgrade infos of a given set of used or unused upgraded segment ids.
+ */
+public class RetrieveUpgradedToSegmentsIdsAction implements TaskAction<List<SegmentUpgradeInfo>>
+{
+  private final String dataSource;
+  private final Set<String> upgradedSegmentIds;
```

Comment:
I think it may be better to leave it as upgradedSegmentIds just to make the difference explicit as the names may seem very similar to a reader unfamiliar with the changes

Reply:
No, I think `upgradedSegmentIds` is actually more confusing, since here it is referring to the parent segment id. "upgradedSegmentId" sounds like the result that you got after upgrading, i.e. the child segment. Using `upgradedFromSegmentId` would not be much better either.

The name of the action itself is enough in this case to distinguish it from the other action.

This action is trying to find the segment IDs to which the given segment IDs were upgraded. So `segmentIds` is adequate.

Reply:
Done